screens/Profile.tsx

```diff
@@ -416,535 +113,304 @@ function ProfileScreenImpl() {
-                        translate("revoke_done_description", {
-                          count: otherInstallations.length,
-                        })
-                      );
-                    }
-                  } catch (error) {
-                    // TODO: Show error feedback to user
-                    sentryTrackError(error);
-                  }
-                },
-              },
-              {
-                id: "delete",
-                title: translate("disconnect_this_account"),
-                titleColor:
-                  Platform.OS === "android"
-                    ? undefined
-                    : dangerColor(colorScheme),
-                action: () => {
-                  setTimeout(() => {
-                    showDisconnectActionSheet(colorScheme);
-                  }, 300);
-                },
-              },
-            ].filter(
-              (i) =>
-                i.id !== "notifications" ||
-                !(notificationsPermissionStatus === "granted")
-            )}
-            title={translate("actions")}
-            style={styles.tableView}
-          />
-          {/* Removing until revoking is available */}
-          {/* <TableView
-            items={[
-              {
-                id: "revoke",
-                title: translate("revoke_other_installations"),
-                titleColor:
-                  Platform.OS === "android"
-                    ? undefined
-                    : dangerColor(colorScheme),
-                action: () => {
-                  setTimeout(() => {
-                    showRevokeActionSheet();
-                  }, 300);
-                },
-              },
-            ]}
-            title={translate("security")}
-            style={styles.tableView}
-          /> */}
+        </HStack>
+      ),
+    },
+    [
+      router,
+      theme,
+      peerAddress,
+      displayName,
+      setPeersStatus,
+      handleChatPress,
+      editMode,
+      handleContextMenuAction,
+      isMyProfile,
+      isBlockedPeer,
+    ]
+  );
 
-          <TableView
-            items={[
-              {
-                id: "version",
-                title: `v${appVersion} (${buildNumber})`,
-              },
-            ]}
-            title={translate("app_version")}
-            style={styles.tableView}
+  return (
+    <Screen preset="fixed" style={themed($container)}>
+      <VStack>
+        <VStack style={themed($section)}>
+          <ContactCard
+            displayName={displayName}
+            userName={formatUsername(userName)}
+            avatarUri={preferredAvatarUri}
+            isMyProfile={isMyProfile}
+            editMode={editMode}
+            onToggleEdit={handleEditProfile}
           />
-        </>
-      )}
-      <View style={{ height: insets.bottom }} />
-    </ScrollView>
+        </VStack>
+
+        {/* Names Section */}
+        {((socials?.userNames?.length ?? 0) > 0 ||
```

Comment:
this feels pretty messy - happy if we want to clean it up later but would be great to start moving away from code like this. 

especially since its so reusable (our handling of arrays of different types of socials and whatnot)